### DISCUSSION TITLE
How to connect remote docker instance using Pulumi?

### DISCUSSION THREAD

**Discussion by madhank93** [pulumi/pulumi#9010] [Pulumi Providers] [answered]
Hi folks,

I have created VM instance in GCP using pulumi and installed docker. And I am trying to connect remote instance of the docker but its getting failed to due to connection establishment (asking for a key verification in a pop up window).

```ts
const remoteInstance = new docker.Provider(
  "remote",
  {
    host: interpolate`ssh://user@${externalIP}:22`,
  },
  { dependsOn: dockerInstallation }
);
```
I can able to run docker containers locally. But want to run the same in VM. The code snippet can be found [here](https://gist.github.com/madhank93/f65bf5119424f91aea23d4abf7dee653)

Any assistance you can provide would be greatly appreciated. Thank you !!!

**Comment by jaxxstorm** [comment, marked as answer]
This is ultimately not a Pulumi problem, but an issue with SSH. In order for the docker provider to connect correctly, you need to be able to connect to the `remoteInstance` with passworddless SSH.

In order to do this, you'll need to make sure your private key uses an [SSH agent](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent) or there's no password on the private key.

You'll also need to disable host key verification. The quickest way to do this is to add the following to `~/.ssh/config`:

```
Host *
    StrictHostKeyChecking no
```

This is not recommended permanently (you should scope the verification to a specific hostname and/or configure verification correctly) but this will allow you to connect to the new instance.

**Reply by madhank93** [reply]
Thank you @jaxxstorm for explaining it in detail.
